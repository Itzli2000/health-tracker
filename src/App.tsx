import { ThemeProvider } from "@/components/theme-provider";
import { ModeToggle } from "@/components/mode-toggle";
import { Button } from "@components/ui/button";
import "./App.css";

function App() {
  return (
    <ThemeProvider>
      <div className="flex flex-col items-center justify-center h-screen bg-background">
        <ModeToggle />
        <h1>Hello World</h1>
        <Button>Click me</Button>
      </div>
    </ThemeProvider>
  );
}

export default App;
