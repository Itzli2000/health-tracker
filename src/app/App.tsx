import { ThemeProvider } from "@/app/providers/theme-provider";
import { BrowserRouter } from "react-router";
import "./App.css";
import AppRoutes from "./components/AppRoutes";

function App() {
  return (
    <ThemeProvider>
      <BrowserRouter>
        <AppRoutes />
      </BrowserRouter>
    </ThemeProvider>
  );
}

export default App;
