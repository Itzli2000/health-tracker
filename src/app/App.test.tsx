import { render, screen } from '@testing-library/react'
import { describe, it, expect } from 'vitest'
import App from './App'

describe('App', () => {
  it('renders hello world heading', () => {
    render(<App />)
    
    const heading = screen.getByRole('heading', { name: /hello world/i })
    expect(heading).toBeInTheDocument()
  })

  it('renders a button with click me text', () => {
    render(<App />)
    
    const button = screen.getByRole('button', { name: /click me/i })
    expect(button).toBeInTheDocument()
  })
})