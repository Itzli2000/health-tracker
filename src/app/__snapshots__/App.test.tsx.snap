// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > should match snapshot 1`] = `
<div>
  <div
    class="group/sidebar-wrapper has-data-[variant=inset]:bg-sidebar flex min-h-svh w-full"
    data-slot="sidebar-wrapper"
    style="--sidebar-width: 14rem; --sidebar-width-icon: 3rem;"
  >
    <div
      class="group peer text-sidebar-foreground hidden md:block"
      data-collapsible=""
      data-side="left"
      data-slot="sidebar"
      data-state="expanded"
      data-variant="sidebar"
    >
      <div
        class="relative w-(--sidebar-width) bg-transparent transition-[width] duration-200 ease-linear group-data-[collapsible=offcanvas]:w-0 group-data-[side=right]:rotate-180 group-data-[collapsible=icon]:w-(--sidebar-width-icon)"
        data-slot="sidebar-gap"
      />
      <div
        class="fixed inset-y-0 z-10 hidden h-svh w-(--sidebar-width) transition-[left,right,width] duration-200 ease-linear md:flex left-0 group-data-[collapsible=offcanvas]:left-[calc(var(--sidebar-width)*-1)] group-data-[collapsible=icon]:w-(--sidebar-width-icon) group-data-[side=left]:border-r group-data-[side=right]:border-l"
        data-slot="sidebar-container"
      >
        <div
          class="bg-sidebar group-data-[variant=floating]:border-sidebar-border flex h-full w-full flex-col group-data-[variant=floating]:rounded-lg group-data-[variant=floating]:border group-data-[variant=floating]:shadow-sm"
          data-sidebar="sidebar"
          data-slot="sidebar-inner"
        >
          <div
            class="flex min-h-0 flex-1 flex-col gap-2 overflow-auto group-data-[collapsible=icon]:overflow-hidden"
            data-sidebar="content"
            data-slot="sidebar-content"
          >
            <div
              class="relative flex w-full min-w-0 flex-col p-2"
              data-sidebar="group"
              data-slot="sidebar-group"
            >
              <div
                class="text-sidebar-foreground/70 ring-sidebar-ring flex h-8 shrink-0 items-center rounded-md px-2 text-xs font-medium outline-hidden transition-[margin,opacity] duration-200 ease-linear focus-visible:ring-2 [&>svg]:size-4 [&>svg]:shrink-0 group-data-[collapsible=icon]:-mt-8 group-data-[collapsible=icon]:opacity-0"
                data-sidebar="group-label"
                data-slot="sidebar-group-label"
              >
                Navegaci√≥n
              </div>
              <div
                class="w-full text-sm"
                data-sidebar="group-content"
                data-slot="sidebar-group-content"
              >
                <ul
                  class="flex w-full min-w-0 flex-col gap-1"
                  data-sidebar="menu"
                  data-slot="sidebar-menu"
                >
                  <li
                    class="group/menu-item relative"
                    data-sidebar="menu-item"
                    data-slot="sidebar-menu-item"
                  >
                    <a
                      class="peer/menu-button flex w-full items-center gap-2 overflow-hidden rounded-md p-2 text-left outline-hidden ring-sidebar-ring transition-[width,height,padding] focus-visible:ring-2 active:bg-sidebar-accent active:text-sidebar-accent-foreground disabled:pointer-events-none disabled:opacity-50 group-has-data-[sidebar=menu-action]/menu-item:pr-8 aria-disabled:pointer-events-none aria-disabled:opacity-50 data-[active=true]:bg-sidebar-accent data-[active=true]:font-medium data-[active=true]:text-sidebar-accent-foreground data-[state=open]:hover:bg-sidebar-accent data-[state=open]:hover:text-sidebar-accent-foreground group-data-[collapsible=icon]:size-8! group-data-[collapsible=icon]:p-2! [&>span:last-child]:truncate [&>svg]:size-4 [&>svg]:shrink-0 hover:bg-sidebar-accent hover:text-sidebar-accent-foreground h-8 text-sm"
                      data-active="true"
                      data-discover="true"
                      data-sidebar="menu-button"
                      data-size="default"
                      data-slot="sidebar-menu-button"
                      data-state="closed"
                      href="/"
                    >
                      <svg
                        aria-hidden="true"
                        class="lucide lucide-house"
                        fill="none"
                        height="20"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                        viewBox="0 0 24 24"
                        width="20"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M15 21v-8a1 1 0 0 0-1-1h-4a1 1 0 0 0-1 1v8"
                        />
                        <path
                          d="M3 10a2 2 0 0 1 .709-1.528l7-5.999a2 2 0 0 1 2.582 0l7 5.999A2 2 0 0 1 21 10v9a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2z"
                        />
                      </svg>
                      <span>
                        Inicio
                      </span>
                    </a>
                  </li>
                  <li
                    class="group/menu-item relative"
                    data-sidebar="menu-item"
                    data-slot="sidebar-menu-item"
                  >
                    <a
                      class="peer/menu-button flex w-full items-center gap-2 overflow-hidden rounded-md p-2 text-left outline-hidden ring-sidebar-ring transition-[width,height,padding] focus-visible:ring-2 active:bg-sidebar-accent active:text-sidebar-accent-foreground disabled:pointer-events-none disabled:opacity-50 group-has-data-[sidebar=menu-action]/menu-item:pr-8 aria-disabled:pointer-events-none aria-disabled:opacity-50 data-[active=true]:bg-sidebar-accent data-[active=true]:font-medium data-[active=true]:text-sidebar-accent-foreground data-[state=open]:hover:bg-sidebar-accent data-[state=open]:hover:text-sidebar-accent-foreground group-data-[collapsible=icon]:size-8! group-data-[collapsible=icon]:p-2! [&>span:last-child]:truncate [&>svg]:size-4 [&>svg]:shrink-0 hover:bg-sidebar-accent hover:text-sidebar-accent-foreground h-8 text-sm"
                      data-active="false"
                      data-discover="true"
                      data-sidebar="menu-button"
                      data-size="default"
                      data-slot="sidebar-menu-button"
                      data-state="closed"
                      href="/dashboard"
                    >
                      <svg
                        aria-hidden="true"
                        class="lucide lucide-layout-dashboard"
                        fill="none"
                        height="20"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                        viewBox="0 0 24 24"
                        width="20"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <rect
                          height="9"
                          rx="1"
                          width="7"
                          x="3"
                          y="3"
                        />
                        <rect
                          height="5"
                          rx="1"
                          width="7"
                          x="14"
                          y="3"
                        />
                        <rect
                          height="9"
                          rx="1"
                          width="7"
                          x="14"
                          y="12"
                        />
                        <rect
                          height="5"
                          rx="1"
                          width="7"
                          x="3"
                          y="16"
                        />
                      </svg>
                      <span>
                        Dashboard
                      </span>
                    </a>
                  </li>
                </ul>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <main
      class="min-h-screen w-full bg-background p-4"
    >
      <div
        class="flex items-center justify-start bg-background gap-2"
      >
        <button
          class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive hover:bg-accent hover:text-accent-foreground dark:hover:bg-accent/50 size-7"
          data-sidebar="trigger"
          data-slot="sidebar-trigger"
        >
          <svg
            aria-hidden="true"
            class="lucide lucide-panel-left"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              height="18"
              rx="2"
              width="18"
              x="3"
              y="3"
            />
            <path
              d="M9 3v18"
            />
          </svg>
          <span
            class="sr-only"
          >
            Toggle Sidebar
          </span>
        </button>
        <button
          aria-expanded="false"
          aria-haspopup="menu"
          class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive border bg-background shadow-xs hover:bg-accent hover:text-accent-foreground dark:bg-input/30 dark:border-input dark:hover:bg-input/50 size-9"
          data-slot="dropdown-menu-trigger"
          data-state="closed"
          id="radix-¬´r2¬ª"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="lucide lucide-sun h-[1.2rem] w-[1.2rem] scale-100 rotate-0 transition-all dark:scale-0 dark:-rotate-90"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="4"
            />
            <path
              d="M12 2v2"
            />
            <path
              d="M12 20v2"
            />
            <path
              d="m4.93 4.93 1.41 1.41"
            />
            <path
              d="m17.66 17.66 1.41 1.41"
            />
            <path
              d="M2 12h2"
            />
            <path
              d="M20 12h2"
            />
            <path
              d="m6.34 17.66-1.41 1.41"
            />
            <path
              d="m19.07 4.93-1.41 1.41"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="lucide lucide-moon absolute h-[1.2rem] w-[1.2rem] scale-0 rotate-90 transition-all dark:scale-100 dark:rotate-0"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20.985 12.486a9 9 0 1 1-9.473-9.472c.405-.022.617.46.402.803a6 6 0 0 0 8.268 8.268c.344-.215.825-.004.803.401"
            />
          </svg>
          <span
            class="sr-only"
          >
            Toggle theme
          </span>
        </button>
      </div>
    </main>
  </div>
</div>
`;
